### PR TITLE
Add "READY" content and examples

### PR DESCRIPTION
**Pull request by dashouse**
- Add additional "READY" content
- Add examples and images
- Add URLs to links within content

**Known issues**
- Typography spacing is off because prose-scope is not yet used
- Inputs can't be resized yet so some examples look weird
- The examples themselves will need a content and code review
- Can't get to the Typography page without entering the URL
- Images aren't styled and don't re-size

### CODE REVIEW

File: src/components/text-input/default.md.njk [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+layout: layout-example.njk
+---
+{% from "input/macro.njk" import govukInput %}
+
+{{ govukInput({
+  "label": {
+    "text": "National insurance number",
+    "hintText": "It’s on your National Insurance card, benefit letter, payslip or P60. For example, ‘QQ 12 34 56 C’."
+  },
+  "id": "input-2",
```
Comment (36degrees) [inline src/components/text-input/default.md.njk]: Again, suggest using 'real' name and id.

File: src/components/text-input/index.md.njk [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## How it works
+
+There are two ways to use the Text input component. You can use HTML or, if you’re using [Nunjucks](https://mozilla.github.io/nunjucks/) or the GOV.UK Prototype Kit, you can use the Text input macro.
+
+{{ example({group: 'components', item: 'text-input', example: 'default', html: true, nunjucks: true, open: true}) }}
+
+### Label text inputs
+
+All text inputs, must have visible labels; placeholder text is not an acceptable replacement for a label as it vanishes when users click on the text input.
```
Comment (36degrees) [inline src/components/text-input/index.md.njk]: Is the punctuation in this sentence correct?? /cc @amyhupe

File: src/patterns/addresses/multiple.md.njk [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+<fieldset class="govuk-c-fieldset">
+  <legend class="govuk-c-fieldset__legend">
+    <h1 class="govuk-heading-xl">What is your address?</h1>
+  </legend>
+
+  <label class="govuk-c-label" for="input-1">
+    Building and street
+  </label>
+  <input class="govuk-c-input" id="input-1" name="test-name" type="text">
```
Comment (36degrees) [inline src/patterns/addresses/multiple.md.njk]: More ids/names

File: src/patterns/confirmation-pages/default.md.njk [hunk 4]
```diff
@@ -4,18 +4,24 @@ layout: layout-example.njk
...
+  <div class="govuk-o-grid">
+    <div class="govuk-o-grid__item govuk-o-grid__item--two-thirds">
+  {{ govukPanel({
+    "titleText": "Application complete",
+    "html": "Your reference number<br><strong>HDJ2123F</strong>"
+  })
+  }}
 
-<p class="govuk-body">We have sent you a confirmation email.</p>
+  <p class="govuk-body">We have sent you a confirmation email.</p>
```
Comment (36degrees) [inline src/patterns/confirmation-pages/default.md.njk]: Can we sort out the indentation in this example?

File: src/patterns/question-pages/postcode.md.njk [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}) }}
+
+<div class="govuk-o-main-wrapper">
+  <div class="govuk-o-grid">
+    <div class="govuk-o-grid__item govuk-o-grid__item--two-thirds">
+      {{ govukInput({
+        "label": {
+          "html": '<h1 class="govuk-heading-xl govuk-!-mb-r4">What is your home postcode?</h1>'
+        },
+        "id": "input-1",
```
Comment (36degrees) [inline src/patterns/question-pages/postcode.md.njk]: More ids/names

File: src/styles/typography/prose-scope.md.njk [hunk 6]
```diff
@@ -0,0 +1,23 @@
+---
+layout: layout-example.njk
+---
+
+<div class="govuk-prose-scope">
+# H1 heading
```
Comment (36degrees) [inline src/styles/typography/prose-scope.md.njk]: I think you'll need to simulate markdown output by marking this up as HTML (without classes) for the scope to style. I think the markdown processor won't touch content inside HTML tags.

File: src/components/button/disabled.md.njk [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+<h3 class="govuk-heading-m">Link button</h3>
+
+{{ govukButton({
+  "text": "Disabled link button",
+  "href": "/",
+  "disabled": true
+}) }}
+
+<h3 class="govuk-heading-m">HTML5 button</h3>
```
[the remark this answers is not in the document]
Comment (dashouse) [reply src/components/button/disabled.md.njk]: What should it be, I have no idea how to explain why this exists...

File: src/components/button/disabled.md.njk [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+{{ govukButton({
+  "text": "Disabled HTML5 button",
+  "href": "/",
+  "element": "button",
+  "disabled": true
+}) }}
```
[the remark this answers is not in the document]
Comment (dashouse) [reply src/components/button/disabled.md.njk]: Fixed

File: src/components/button/disabled.md.njk (hunk 7 above)
[the remark this answers is not in the document]
Comment (joelanman) [reply src/components/button/disabled.md.njk]: the main difference is `button` can contain html, `input` cannot - can't remember why we can't just use `button`, @36degrees ?

File: src/components/button/start.md.njk [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+<h3 class="govuk-heading-m">Input Start button</h3>
+
+{{ govukButton({
+  "text": "Start now button",
+  "classes": "govuk-c-button--start"
+}) }}
+
+<h3 class="govuk-heading-m">Link Start button</h3>
+
+{{ govukButton({
```
[the remark this answers is not in the document]
Comment (dashouse) [reply src/components/button/start.md.njk]: Good spot - fixed

File: src/components/fieldset/index.md.njk [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+On [Question pages](../../patterns/question-pages) containing a group of inputs, including the question as the legend helps users of screenreaders to understand that the inputs are all related to that question.
+
+Include any general help text which is important for filling in the form and can’t be written as [hint text](Link to hint text section of help users Fill in a form) in the legend, but try to keep it as concise as possible.
+
+## Research on this component
+
+If you’ve used this component, [get in touch](Anchor link to Get in touch panel at bottom of page) to share your user research findings.
+
+
```
[the remark this answers is not in the document]
Comment (dashouse) [reply src/components/fieldset/index.md.njk]: Fixed

File: src/patterns/confirmation-pages/default.md.njk (hunk 4 above)
Comment (dashouse) [reply src/patterns/confirmation-pages/default.md.njk]: If I indent this properly all of the text turns into a code block

File: src/styles/typography/prose-scope.md.njk (hunk 6 above)
Comment (dashouse) [reply src/styles/typography/prose-scope.md.njk]: Fixed

File: src/patterns/confirmation-pages/default.md.njk (hunk 4 above)
Comment (joelanman) [reply src/patterns/confirmation-pages/default.md.njk]: can you do what I did with colours, and just make this a .njk file, not a .md.njk? then it won't get the code block issue

Comment (36degrees) [reply src/patterns/confirmation-pages/default.md.njk]: Yeah, there's probably no reason for any of the examples to use the `.md` extension as none of them contain markdown…

Comment (dashouse) [reply src/patterns/confirmation-pages/default.md.njk]: Fixed

File: src/components/button/disabled.md.njk (hunk 7 above)
[the remark this answers is not in the document]
Comment (36degrees) [reply src/components/button/disabled.md.njk]: There's some funky behaviour in IE8 and below (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Notes) which is why we've opted for input being the default unless you try to put HTML inside it.

However, as part of the work I've been doing revisiting the button I've found that `<input>`s also don't allow pseudo elements, which means some tweaks we make to allow you to click the very top of the button don't work – so I've been wondering about changing it to default to `<button>` with a implementation note, whilst continuing to provide input as an option for those that need it.

File: src/components/button/primary.njk [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+{{ govukButton({
+  "text": "Save and continue"
+}) }}
+
+<h3 class="govuk-heading-m">Link button</h3>
+
+{{ govukButton({
+  "text": "Link button",
+  "href": "/"
```
Comment (hannalaakso) [inline src/components/button/primary.njk]: Only a small thing but instead of `/` can we use `#` as currently clicking on the button opens the homepage which is a bit weird.

File: src/components/button/start.njk [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+  "text": "Start now button",
+  "classes": "govuk-c-button--start"
+}) }}
+
+<h3 class="govuk-heading-m">Link Start button</h3>
+
+{{ govukButton({
+  "text": "Start now link button",
+  "href": "/",
+  "classes": "govuk-c-button--start"
```
Comment (hannalaakso) [inline src/components/button/start.njk]: Same thing about using # instead of /

File: src/components/checkboxes/default.njk [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    {
+      "value": "irish",
+      "text": "Irish"
+    },
+    {
+      "value": "other",
+      "text": "Citizen of another country"
+    }
+  ]
+}) }}
```
Comment (hannalaakso) [inline src/components/checkboxes/default.njk]: A more of a general question but should we have more variants of the checkboxes here? Such as having a disabled and "None" option etc. Or maybe that's coming later.

File: src/components/radios/index.md.njk [hunk 14]
```diff
@@ -19,19 +21,19 @@ Don’t use the Radios component if users might need to select more than one opt
 
 ## How it works
```
Comment (hannalaakso) [inline src/components/radios/index.md.njk]: Not part of this PR but might be worth fixing here anyway: there's a mistake closing a link on line 20 of this file `[Checkboxes component](/components/checkboxes/] `

File: src/components/radios/default.md.njk [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+layout: layout-example.njk
+---
+
+{% from 'radio/macro.njk' import govukRadio %}
+
+{{ govukRadio({
+  "idPrefix": "example",
+  "name": "example",
+  "fieldset": {
+    "legendHtml": "<h1 class=\govuk-heading-xl\>Have you changed your name?</h1>",
```
Comment (alex-ju) [inline src/components/radios/default.md.njk]: I would've probably used a mix of single and double quotes `'legendHtml': '<h1 class="govuk-heading-xl">Have you changed your name?</h1>'`, but it's just a personal option.

File: src/components/radios/index.md.njk [hunk 16]
```diff
@@ -25,13 +27,13 @@ There are two ways to use the Radios component. You can use HTML or, if you are
 
 When there are more than two options, radios should be stacked, like so:
 
-[Example of stacked radios]
+{{ example({group: 'components', item: 'radios', example: 'default', html: true, nunjucks: true, open: true}) }}
```
Comment (alex-ju) [inline src/components/radios/index.md.njk]: I will extend the example tab container id to allow the same example being placed on the same page with and without code blocks - discussed this problem with @dashouse offline.

File: src/styles/typography/index.md.njk [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+Introduce bulleted lists with a lead in line ending in a colon. Start each item with a lowercase letter, and don’t use a full stop at the end.
+
+{{ example({group: 'styles', item: 'typography', example: 'list-bullet', html: true, nunjucks: true, open: false}) }}
+
+### Numbered lists
+
+Use numbered lists instead of bulleted lists when the order of the items is relevant.
+
+You don’t need to use a lead-in line for numbered lists. Items in a numbered list should end in a full stop because each should be a complete sentence.
+
```
Comment (hannalaakso) [inline src/styles/typography/index.md.njk]: The examples for this do not end in a full stop or aren't a complete sentence.

File: src/components/skip-link/default.md.njk [hunk 18]
```diff
@@ -0,0 +1,11 @@
+---
+layout: layout-example.njk
+---
+<p class="govuk-body">To view the Skip link component click inside this example and press tab</p>
+
```
Comment (alex-ju) [inline src/components/skip-link/default.md.njk]: I would remove this empty line because introduces an extra line between `<p>` and `<a>` on the HTML example
screen shot 2018-01-04 at 16 58 39

File: src/patterns/fill-in-a-form/index.md.njk [hunk 19]
```diff
@@ -111,6 +111,35 @@ Use the Details component to make your page easier to scan, but don’t hide hel
...
+
+
+
+### Using progress indicators
+
+Start by testing your form without a progress indicator to see if it’s simple enough that users don’t need them.
+
+Try improving the order, type or number of questions before adding a progress indicator. If people still have difficulty, try adding a simple step or question indicator like this one.
+
+{{ example({group: 'patterns', item: 'fill-in-a-form', example: 'progress', html: true, nunjucks: true, open: false}) }}
```
Comment (alex-ju) [inline src/patterns/fill-in-a-form/index.md.njk]: This example doesn't contain any Nunjucks syntax. Shall we remove the Nunjucks tab for it?

**Comment by alex-ju** [comment]
We still lacking a few new lines at the end of the file, but I'm happy to help with that.

File: src/components/button/primary.njk (hunk 11 above)
Comment (dashouse) [reply src/components/button/primary.njk]: Fixed

File: src/components/button/start.njk (hunk 12 above)
Comment (dashouse) [reply src/components/button/start.njk]: Fixed

File: src/components/checkboxes/default.njk (hunk 13 above)
Comment (dashouse) [reply src/components/checkboxes/default.njk]: Yes I think this will be a content decision - We'll do a bigger content and examples review once everything is in.

File: src/components/radios/index.md.njk (hunk 14 above)
Comment (dashouse) [reply src/components/radios/index.md.njk]: Fixed

File: src/styles/typography/index.md.njk (hunk 17 above)
Comment (dashouse) [reply src/styles/typography/index.md.njk]: Great spot, thanks!

File: src/patterns/fill-in-a-form/index.md.njk (hunk 19 above)
Comment (alex-ju) [reply src/patterns/fill-in-a-form/index.md.njk]: There is a bug on the tabs right now that needs investigating. This example will be updated after fixing the tabs problem.

Comment (dashouse) [reply src/patterns/fill-in-a-form/index.md.njk]: We'll need to fix this once we've had a look at the examples configuration
